flink/src/main/java/org/apache/iceberg/flink/data/vectorized/VectorizedFlinkOrcReaders.java

```diff
@@ -0,0 +1,381 @@
+              public DecimalData getDecimal(int i, int precision, int scale) {
+                return (DecimalData) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case DATE:
+            return new org.apache.flink.table.data.vector.IntColumnVector() {
+              @Override
+              public int getInt(int i) {
+                return (int) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case TIME:
+            return new org.apache.flink.table.data.vector.IntColumnVector() {
+              @Override
+              public int getInt(int i) {
+                return (int) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case TIMESTAMP:
+            return new org.apache.flink.table.data.vector.TimestampColumnVector() {
+              @Override
+              public TimestampData getTimestamp(int i, int precision) {
+                return (TimestampData) constant;
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+          case STRING:
+            return new org.apache.flink.table.data.vector.BytesColumnVector() {
+              @Override
+              public Bytes getBytes(int i) {
+                BinaryStringData str = (BinaryStringData) constant;
+                return new Bytes(str.toBytes(), 0, str.getSizeInBytes());
+              }
+
+              @Override
+              public boolean isNullAt(int i) {
+                return false;
+              }
+            };
+
+
+          default:
+            throw new UnsupportedOperationException("Unsupported data type for constant.");
+        }
+      } else {
+        throw new UnsupportedOperationException("ConstantColumnVector only supports primitives.");
+      }
+    }
+
+    private static class FlinkRowColumnVector implements RowColumnVector {
+
+      private ColumnVector[] fieldVectors;
+      private StructColumnVector structVector;
+
+      FlinkRowColumnVector(ColumnVector[] fieldVectors,
+                           StructColumnVector structVector) {
+        this.fieldVectors = fieldVectors;
+        this.structVector = structVector;
+      }
+
+      @Override
+      public ColumnarRowData getRow(int i) {
+        VectorizedColumnBatch vectorizedColumnBatch = new VectorizedColumnBatch(fieldVectors);
+        return new ColumnarRowData(vectorizedColumnBatch, i);
+      }
+
+      @Override
+      public boolean isNullAt(int i) {
+        return structVector.isNull[i];
+      }
+
+      public ColumnVector[] getFieldVectors() {
+        return fieldVectors;
+      }
+    }
```

Comment:
This class should be an inner class of the outside VectorizedFlinkOrcReaders , rather than an inner static class of StructConverter ?